test-ts/proxy-deploy.test.ts

```diff
@@ -2,79 +2,298 @@ import { expect } from "chai";
 import hre from "hardhat";
 import { StakedCelo__factory } from "../typechain-types/factories/StakedCelo__factory";
 import { StakedCelo } from "../typechain-types/StakedCelo";
+import { Account } from "../typechain-types/Account";
+import { RebasedStakedCelo } from "../typechain-types/RebasedStakedCelo";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { randomAddress, randomSigner } from "./utils";
 import { parseUnits } from "ethers/lib/utils";
+import { Account__factory } from "../typechain-types/factories/Account__factory";
+import { Manager__factory } from "../typechain-types/factories/Manager__factory";
+import { RebasedStakedCelo__factory } from "../typechain-types/factories/RebasedStakedCelo__factory";
+import { Manager } from "../typechain-types/Manager";
 
 describe("Contract deployed via proxy", () => {
-  let StakedCelo: StakedCelo__factory;
   let owner: SignerWithAddress;
+  let deployer: SignerWithAddress;
 
-  before(async () => {
+  beforeEach(async () => {
     owner = await hre.ethers.getNamedSigner("owner");
+    deployer = await hre.ethers.getNamedSigner("deployer");
+    console.error("Test owner ", owner.address);
   });
 
-  beforeEach(async () => {
-    await hre.deployments.fixture("TestStakedCelo");
-    StakedCelo = await hre.ethers.getContractFactory("StakedCelo");
+  describe("StakedCelo", () => {
```

Comment:
Seems like this test is setup to verify the proxy pattern used. 
Since all contracts follow the same proxy pattern, is it really necessary to repeat the same test for each?

Reply:
I would argue that this is exactly the reason why we want the test - to make sure that really every contract is following proxy pattern

Reply:
Agreed, we want to test all proxies behave correctly. However, the common code could be abstracted away to a function, then run it passing each of the contracts separately (just to clean up and deduplicate code).

Reply:
Agreed - tests are abstracted